pkg/util/kernel/download_headers.go

```diff
@@ -0,0 +1,136 @@
+// +build linux
+
+package kernel
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/nikos/apt"
+	"github.com/DataDog/nikos/cos"
+	"github.com/DataDog/nikos/rpm"
+	"github.com/DataDog/nikos/types"
+	"github.com/DataDog/nikos/wsl"
+)
+
+// customLogger is a wrapper around our logging utility which allows nikos to use our logging functions
+type customLogger struct{}
+
+func (c customLogger) Debug(args ...interface{})                 { log.Debug(args...) }
+func (c customLogger) Info(args ...interface{})                  { log.Info(args...) }
+func (c customLogger) Warn(args ...interface{})                  { log.Warn(args...) }
+func (c customLogger) Error(args ...interface{})                 { log.Error(args...) }
+func (c customLogger) Debugf(format string, args ...interface{}) { log.Debugf(format, args...) }
+func (c customLogger) Infof(format string, args ...interface{})  { log.Infof(format, args...) }
+func (c customLogger) Warnf(format string, args ...interface{})  { log.Warnf(format, args...) }
+func (c customLogger) Errorf(format string, args ...interface{}) { log.Errorf(format, args...) }
+
+var _ types.Logger = customLogger{}
+
+// downloadHeaders attempts to download kernel headers & place them in headerDownloadDir
+func downloadHeaders(headerDownloadDir string) ([]string, error) {
+	var (
+		target    types.Target
+		backend   types.Backend
+		outputDir string
+		err       error
+	)
+
+	// Before downloading new kernel headers, we'll delete anything we previously downloaded (if it exists)
+	deleteDirContents(headerDownloadDir)
```

Comment:
This seems especially dangerous since this value can be user provided and we are doing effectively a `rm -rf` on this directory. 

Do we need to delete previous contents? If not, can we restrict it to directories we created somehow?

Reply:
We don't need to delete previous contents, I just thought it would be an easy way to clean up after ourselves in case we end up needing to download multiple kernel headers over some period of time (ie if the user upgrades their kernel version or something like that). I'll change this so that we only delete the contents if `headerDownloadDir` is the default value, that way we know we won't delete anything dangerous - though I'm not against removing this behaviour entirely if you think it's not worth doing this.

Reply:
I think the safest options are:
1. Don't do it at all.
2. Use a unique prefix on the directory names so we know they came from the system-probe, and only delete those.

I think you can remove it for now, and we can revisit it later. Can you add a card for that?

Reply:
Sounds good to me. & will do!